projects/dashboards/gfbio/src/app/gfbio-collection/gfbio-collection-dialog.component.ts

```diff
@@ -0,0 +1,75 @@
+import {Component, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+import {LayerCollectionDict, LayerCollectionItemDict, LayerCollectionService, ProjectService} from '@geoengine/core';
+import {map, mergeMap} from 'rxjs/operators';
+import {Observable, of} from 'rxjs';
+
+enum LayerStatus {
+    Ok = 'ok',
+    Unavailable = 'unavailable',
+    Error = 'error',
+}
+
+@Component({
+    selector: 'geoengine-gfbio-basket-dialog',
+    templateUrl: './gfbio-collection-dialog.component.html',
+    styleUrls: ['./gfbio-collection-dialog.component.scss'],
+})
+export class GfBioCollectionDialogComponent {
+    collection: LayerCollectionDict;
+    hasLayers$: Observable<boolean>;
+
+    constructor(
+        private readonly projectService: ProjectService,
+        private readonly layerService: LayerCollectionService,
+        private dialogRef: MatDialogRef<GfBioCollectionDialogComponent>,
+        @Inject(MAT_DIALOG_DATA) private config: {result: LayerCollectionDict},
+    ) {
+        this.collection = config.result;
+        this.hasLayers$ = this.projectService.getLayerStream().pipe(map((layers) => layers.length > 0));
+    }
+
+    layerStatus(item: LayerCollectionItemDict): LayerStatus {
+        const status = item.properties.find(([a, _]) => a === 'status');
+
+        if (status) {
+            return this.statusFromString(status[1]);
+        } else {
+            return LayerStatus.Error;
+        }
+    }
+
+    appendLayers(): void {
+        const filteredCollection = this.collection;
+        this.collection.items = this.getOkLayers();
+
+        this.layerService.addCollectionLayersToProject(filteredCollection);
+        this.dialogRef.close();
+    }
+
+    replaceLayers(): void {
+        const filteredCollection = this.collection;
+        this.collection.items = this.getOkLayers();
+
+        this.projectService
+            .clearLayers()
+            .pipe(mergeMap(() => of(this.layerService.addCollectionLayersToProject(filteredCollection))))
+            .subscribe();
+        this.dialogRef.close();
+    }
+
+    private getOkLayers(): LayerCollectionItemDict[] {
+        return this.collection.items.filter((item) => this.layerStatus(item) === LayerStatus.Ok);
+    }
+
+    private statusFromString(status: string): LayerStatus {
+        switch (status) {
+            case 'ok':
+                return LayerStatus.Ok;
+            case 'unavailable':
+                return LayerStatus.Unavailable;
+            default:
+                return LayerStatus.Error;
```

Comment:
Is it wise to log an error if the status is not error, but we fall back to it?
Falling back to an error as it is now makes sense, since there would be something wrong anyway.

Reply:
What does it achieve to log this? We could introduce a fourth type "Status N/A", then users could report this as a bug. Because the only way that the status is missing in the layer properties is if there is a bug in the data provider.

Reply:
The intermediate way would be to log/output it that this should be reported. Then, we don't need another handling than `Error`. It will be visible as a notification and in the log menu.

Reply:
Do you mean notification with our NotificationService? 
What if there are multiple elements with error? One notification per error?
To whom should it be reported? Just like "Contact the administrator"?

Reply:
Yeah, I guess. Just plain and simple report what happened.

Reply:
68aaaa107ab1538aa785539828ccfccf457074de

you can only see one error as toast, but in the log you can see the rest